p2p/protocols/protocol.go

```diff
@@ -336,6 +359,19 @@ func (p *Peer) handleIncoming(handle func(ctx context.Context, msg interface{})
 		return errorf(ErrDecode, "<= %v: %v", msg, err)
 	}
 
+	//if the accounting hook is set, call it
+	if p.spec.Hook != nil {
+		if wmsg.Size != uint32(len(wmsg.Payload)) {
+			errMsg := "Advertised message size and payload length don't match"
+			log.Warn(errMsg)
+			p.Drop(errors.New(errMsg))
```

Comment:
Why are we dropping a peer here? I don't think this is responsibility of the `p2p/protocols` package.

Reply:
it is part of the contract with the hook: the consequence of returning an error is breaking. 
Do not return an error if you dont want this behaviour